### ISSUE TITLE
Opt-out of sudo - allow sudo=False to work with --sudo

### ISSUE THREAD

**Issue by pylang** [Fizzadar/pyinfra#580]
## Is your feature request related to a problem? Please describe

More than 2/3s of my tasks use sudo.  However, passing `sudo=True` to all operations that use them (e.g. anything uses `apt.packages()`) is a lot of copy/pasting.

## Describe the solution you'd like

- Is there a DRYer way of using sudo?
- Can I turn off warnings at runtime that complain about using `sudo=False` with `--sudo`?
- Can operations have a way to opt-out of sudo when `--sudo` is passed, without warnings or exceptions?

<details>

The workaround I have is to always pass the `--sudo` flag at runtime.  

```bash
> pyinfra --sudo --use-sudo-password @local deploy_*.py
```

This handles most tasks, except those few that do not use sudo.  In those rarer cases, I hardcode `sudo=False` to the few operations that need it.

*Issue 1*

Lately, I've noticed there is a warning that emits when I run the latter command:

        Invalid auth argument: cannot use `use_sudo_password=('pyinfra-sudo-askpass', 'password')` without `sudo`

This is complaining about my passing `sudo=False`  together with `--sudo`, and it complains a lot. 

Can I turn this warning off?

*Issue 2*

I see there is some thought about [throwing an exception instead of a warning][0].  An exception which would wreck this workaround and leave me without solutions to my original problem.  

I request not to throw an exception here until you have a way to let operations opt-out of  `--sudo`.

</details>

 [0]: https://github.com/Fizzadar/pyinfra/blob/0bea34e774ce40cbab6049bc754d0b29d11263fd/pyinfra/api/connectors/util.py#L212

**Comment by Fizzadar** [comment]
@pylang to fix the default argument you could set sudo as a config variable, ie:

```py
SUDO = True

server.shell(...
```

This would avoid needing to pass it as a CLI argument every invokation.

Regarding the warnings - you raise an issue here where the global configured defaults (CLI arg or config) for an operation may differ from operation specific auth arguments and should not show any warning in those cases. So the warning check neeeds to identify if the invalid argument combination is an accident or intentional (as in this case). This kind of side effect is part of why I went down the warning route because it's hard to get this kind of thing right, and an exception would be much more painful for everyone!

**Comment by Fizzadar** [comment]
Pushed https://github.com/Fizzadar/pyinfra/commit/9f3255934be976e1fbc92276e9b00d0f24043533 which disables the warnings in the described scenario, and just released in `v1.4.1`! Going to close this issue now as I believe this is everything resolved.
